tofu/imas2tofu/_core.py

```diff
@@ -2243,52 +2274,129 @@ def to_Config(self, Name=None, occ=None, indDescription=None, plot=True):
             occ = occ[0]
             indoc = np.nonzero(self._dids[ids]['occ'] == occ)[0][0]
 
-            wall = self._dids[ids]['ids'][indoc]
-            units = wall.description_2d[indDescription].limiter.unit
-            nunits = len(units)
-
-            if nunits == 0:
-                msg = "There is no limiter unit stored !\n"
-                msg += "The required 2d description is empty:\n"
-                ms = "len(idd.%s[occ=%s].description_2d"%(ids,str(occ))
-                msg += "%s[%s].limiter.unit) = 0"%(ms,str(indDescription))
-                raise Exception(msg)
-
-            if Name is None:
-                Name = wall.description_2d[indDescription].type.name
-                if Name == '':
-                    Name = 'imas wall'
+            wall = self._dids[ids]['ids'][indoc].description_2d[description_2d]
+            kwargs = dict(Exp=Exp, Type='Tor')
 
             import tofu.geom as mod
 
-            lS = [None for _ in units]
-            kwargs = dict(Exp=Exp, Type='Tor')
-            for ii in range(0,nunits):
-                poly = np.array([units[ii].outline.r, units[ii].outline.z])
-
-                if units[ii].phi_extensions.size > 0:
-                    pos, extent =  units[ii].phi_extensions.T
+            # Get vessel
+            nlim = len(wall.limiter.unit)
+            nmob = len(wall.mobile.unit)
+            onelimonly = False
+            try:
+                if len(wall.vessel.unit) != 1:
```

Comment:
There should be only one vessel in the IMAS ids (corresponds to tofu call Ves : a StructIn sub-class)